QMLComponents/controls/jaspcontrol.cpp

```diff
@@ -248,6 +255,13 @@ void JASPControl::componentComplete()
 
 	if (_form)
 		connect(this, &JASPControl::boundValueChanged, _form, &AnalysisForm::boundValueChangedHandler);
+
+	//capture focus reason
+	for (QQuickItem* childItem : childItems())
+	{
+		if (!qobject_cast<JASPControl*>(childItem))
+			childItem->installEventFilter(this);
+	}
```

Comment:
Why all/only child items that are not JASPControl should get this event filter. Should it be only the inner control? Maybe the right control that receives the focus is not a direct child?